does not build on Mojave with go 1.12.2

I am not able to build the current head or the last release:

```
$make
GOOS=linux GOARCH=amd64 go build -o cpfw-login_amd64 .
# _/Users/martin/workspace/cpfw-login
./http.go:21:102: cannot use "password" (type string) as type http.SameSite in field value
./http.go:21:122: cannot use []string literal (type []string) as type string in field value
./http.go:21:122: too few values in &http.Cookie literal
./http.go:22:94: cannot use "password" (type string) as type http.SameSite in field value
./http.go:22:114: cannot use []string literal (type []string) as type string in field value
./http.go:22:114: too few values in &http.Cookie literal
make: *** [cpfw-login_amd64] Error 2
```

```
$make cpfw-login_darwin
GOOS=darwin go build -o cpfw-login_darwin .
# _/Users/martin/workspace/cpfw-login
./http.go:21:102: cannot use "password" (type string) as type http.SameSite in field value
./http.go:21:122: cannot use []string literal (type []string) as type string in field value
./http.go:21:122: too few values in &http.Cookie literal
./http.go:22:94: cannot use "password" (type string) as type http.SameSite in field value
./http.go:22:114: cannot use []string literal (type []string) as type string in field value
./http.go:22:114: too few values in &http.Cookie literal
make: *** [cpfw-login_darwin] Error 2
```

With go version go1.12.2 darwin/amd64

--- a/cookie_new.go
+++ b/cookie_new.go
@@ -1,0 +1,16 @@
+// +build go1.12
+
+package main
+
+import (
+	"net/http"
+	"time"
+)
+
+func loginTypeCookie(host string, time time.Time) *http.Cookie {
+	return &http.Cookie{"cpnacportal_login_type", "password", "/", host, time, "", 0, true, false, http.SameSiteLaxMode, "password", []string{"password"}}
+}
+
+func userCookie(host string, time time.Time, user string) *http.Cookie {
+	return &http.Cookie{"cpnacportal_username", user, "/", host, time, "", 0, true, false, http.SameSiteLaxMode, "password", []string{user}}
+}

--- a/cookie_old.go
+++ b/cookie_old.go
@@ -1,0 +1,16 @@
+// +build !go1.12
+
+package main
+
+import (
+	"net/http"
+	"time"
+)
+
+func loginTypeCookie(host string, time time.Time) *http.Cookie {
+	return &http.Cookie{"cpnacportal_login_type", "password", "/", host, time, "", 0, true, false, "password", []string{"password"}}
+}
+
+func userCookie(host string, time time.Time, user string) *http.Cookie {
+	return &http.Cookie{"cpnacportal_username", user, "/", host, time, "", 0, true, false, "password", []string{user}}
+}

--- a/http.go
+++ b/http.go
@@ -18,8 +18,8 @@ func initCookies(uri, user string) *cookiejar.Jar {
 	cookies := make([]*http.Cookie, 2)
 	t := time.Now()
 	t = t.Add(5 * time.Hour)
-	cookies[0] = &http.Cookie{"cpnacportal_login_type", "password", "/", u.Host, t, "", 0, true, false, "password", []string{"password"}}
-	cookies[1] = &http.Cookie{"cpnacportal_username", user, "/", u.Host, t, "", 0, true, false, "password", []string{user}}
+	cookies[0] = &http.Cookie{"cpnacportal_login_type", "password", "/", u.Host, t, "", 0, true, false, http.SameSiteLaxMode, "password", []string{"password"}}
+	cookies[1] = &http.Cookie{"cpnacportal_username", user, "/", u.Host, t, "", 0, true, false, http.SameSiteLaxMode, "password", []string{user}}
 	// log.Printf("cookies: %v", cookies)
 	jar.SetCookies(u, cookies)
 	return jar

--- a/http.go
+++ b/http.go
@@ -18,8 +18,8 @@ func initCookies(uri, user string) *cookiejar.Jar {
 	cookies := make([]*http.Cookie, 2)
 	t := time.Now()
 	t = t.Add(5 * time.Hour)
-	cookies[0] = &http.Cookie{"cpnacportal_login_type", "password", "/", u.Host, t, "", 0, true, false, http.SameSiteLaxMode, "password", []string{"password"}}
-	cookies[1] = &http.Cookie{"cpnacportal_username", user, "/", u.Host, t, "", 0, true, false, http.SameSiteLaxMode, "password", []string{user}}
+	cookies[0] = loginTypeCookie(u.Host, t)
+	cookies[1] = userCookie(u.Host, t, user)
 	// log.Printf("cookies: %v", cookies)
 	jar.SetCookies(u, cookies)
 	return jar

